scheduler/scheduler.cpp

```diff
@@ -722,7 +723,41 @@ static time_t prune_servers()
         ++it;
     }
 
-    for (it = css.begin(); it != css.end();) {
+    list<CompileServer *> css_all;
+    css_all.insert(css_all.begin(), css.begin(), css.end());
+    css_all.insert(css_all.begin(), css_temp_hold.begin(), css_temp_hold.end());
+
+    for (it = css_all.begin(); it != css_all.end();) {
+        if ((*it)->incomingConnectionTestRequired())
+        {
+            if (find(css.begin(), css.end(), (*it)) != css.end() && !(*it)->incomingConnectionAccepted())
+            {
+                trace() << "Client (" << (*it)->nodeName() <<
+                    " " << (*it)->name <<
+                    ":" << (*it)->remotePort() <<
+                    ") connected but is not able to accept incoming connections." << endl;
+                //move or copy cs(old) to list css_temp_hold;(copy means that we can still do
+                //disconnection checks and removals from the css list)
+                CompileServer *hold = *it;
+                css_temp_hold.push_back(hold);
+                css.remove(hold);
```

Comment:
Note that we need to get the iterator from find (currently in the if and lost), not it as I wrote.